client/src/App.tsx

```diff
@@ -210,6 +216,90 @@ function App() {
 		}
 	}, []);
 
+	const activateFireworks = () => {
+		var duration = 2 * 1000;
+		var animationEnd = Date.now() + duration;
+		var defaults = { startVelocity: 30, spread: 360, ticks: 60, zIndex: 0 };
+
+		function randomInRange(min: number, max: number) {
```

Comment:
avoid defining functions inside other functions, should move this outside, ideally into a util.ts file with other general purpose functions like this